File: Easy/istanza/meta_istanzaimm_segrinview/meta_istanzaimm_segrinview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_istanzaimm_segrinview
{
    public class Meta_istanzaimm_segrinview : Meta_easydata 
	{
        public Meta_istanzaimm_segrinview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "istanzaimm_segrinview") {
				Name = "Istanze di rinnovo della iscrizione";
			EditTypes.Add("imm_segrin");
			ListingTypes.Add("imm_segrin");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idistanza","idreg_studenti"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "imm_segrin": {
						DescribeAColumn(T, "istanza_aa", "Anno accademico", nPos++);
						DescribeAColumn(T, "registrystudenti_title", "Studente", nPos++);
						DescribeAColumn(T, "istanza_data", "Data", nPos++);
						if (T.Columns.Contains("istanza_data")) T.Columns["istanza_data"].ExtendedProperties["format"] = "g";
						DescribeAColumn(T, "statuskind_title", "Status", nPos++);
						DescribeAColumn(T, "istanza_protnumero", "Numero di protocollo", nPos++);
						DescribeAColumn(T, "istanza_protanno", "Anno di protocollo", nPos++);
						DescribeAColumn(T, "didprog_title", "Didattica programmata", nPos++);
						DescribeAColumn(T, "didprogcurr_title", "Curriculum", nPos++);
						DescribeAColumn(T, "didprogori_title", "Corso e orientamento", nPos++);
						DescribeAColumn(T, "istanza_imm_parttime", "Iscrizione Part-Time", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		public override string GetStaticFilter(string ListingType) {
			switch (ListingType) {
				case "imm_segrin": {
						return "istanza_idistanzakind = 15";
						break;
					}
				//$GetStaticFilter$
			}
			return base.GetStaticFilter(ListingType);
		}
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/istanza/meta_istanzaimm_segrinview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_istanzaimm_segrinview() {
        MetaData.apply(this, ["istanzaimm_segrinview"]);
        this.name = 'meta_istanzaimm_segrinview';
    }

    meta_istanzaimm_segrinview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_istanzaimm_segrinview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'imm_segrin':
						this.describeAColumn(table, 'aa', 'Anno accademico', null, 1000, 9);
						this.describeAColumn(table, 'registrystudenti_title', 'Studente', null, 2300, 101);
						this.describeAColumn(table, 'istanza_data', 'Data', 'g', 3000, null);
						this.describeAColumn(table, 'statuskind_title', 'Status', null, 8200, 50);
						this.describeAColumn(table, 'didprogcurr_title', 'Curriculum', null, 52200, 256);
						this.describeAColumn(table, 'didprogori_title', 'Corso e orientamento', null, 53200, 256);
						this.describeAColumn(table, 'iscrizione_aa', 'Anno accademico Iscrizione', null, 55100, 9);
						this.describeAColumn(table, 'iscrizione_anno', 'Anno di corso Iscrizione', null, 55300, null);
						this.describeAColumn(table, 'iscrizione_iddidprog', 'Didattica programmata Iscrizione', null, 55500, null);
						this.describeAColumn(table, 'istanza_imm_parttime', 'Iscrizione Part-Time', null, 57000, null);
						this.describeAColumn(table, 'istanza_protnumero', 'Numero di protocollo', null, 61000, null);
						this.describeAColumn(table, 'istanza_protanno', 'Anno di protocollo', null, 62000, null);
//$objCalcFieldConfig_imm_segrin$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["iddidprog", "idistanza", "idcorsostudio", "idistanzakind", "idreg_studenti"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

        });

    window.appMeta.addMeta('istanzaimm_segrinview', new meta_istanzaimm_segrinview('istanzaimm_segrinview'));

	}());
